src/prefect/core/flow.py

```diff
@@ -215,6 +217,12 @@ def __repr__(self) -> str:
     def __iter__(self) -> Iterable[Task]:
         yield from self.sorted_tasks()
 
+    def __getstate__(self) -> Dict[str, Any]:
```

Comment:
This is the only thing that I'm waffling over about this PR --> I 100% understand why you did it / what it solves for, but it feels "heavy" to me for some reason.  Gonna keep thinking on this.

Reply:
Because `flow.__exit__` deletes the generator, this is really just a redundant guard against an unlikely case (pickling when within a flow context). I'll pull it out.